bokeh/models/widgets/buttons.py

```diff
@@ -26,7 +26,7 @@ class AbstractButton(Widget):
     An optional image appearing to the left of button's text.
     """)
 
-    type = Enum(ButtonType, help="""
+    button_type = Enum(ButtonType, help="""
```

Comment:
On Python-level I can add deprecation code to ease the transition. It's just bokehjs where this has to be `!= "type"`.